plugins/strategy/action.go

```diff
@@ -86,3 +125,40 @@ func (a *Action) pushReason(r string) {
 	a.Meta[metaKeyReasonHistory] = history
 	a.Reason = r
 }
+
+// PreemptAction determines which Action should take precedence.
+//
+// The result is based on the scaling direction and count. The order of
+// precedence for the scaling directions is defined by the order in which they
+// are declared in the above enum.
+//
+// If the scaling direction is the same, the priority is given to the safest
+// option, where safest is defined as lowest impact in the underlying
+// infrastructure:
+//
+//   * ScaleDirectionUp: Action with highest count
+//   * ScaleDirectionDown: Action with highest count
+func PreemptAction(a *Action, b *Action) *Action {
+	if a == nil {
+		return b
+	}
+
+	if b == nil {
+		return a
+	}
+
+	if b.Direction > a.Direction {
```

Comment:
the direction checks have the assumption that they are working from the same initial count... that's true, correct?

Reply:
"yes". It's a very soft lower-case yes because it actually it isn't.

The final action is based on two values: current count and metric query result. Since each check has a different query associated we will have multiple results there and so there's a small time difference. 

We could retrieve current count just once, but then the metric is still not guarantee to be read at the time that count is valid.

The ideal scenario would be to pass a timestamp when getting target count and metric, but the Nomad APM doesn't have historical data.

So I tried to bundle these calls as close as possibles, but there isn't any guarantee that all checks are seeing the same data.